packages/roosterjs-editor-core/lib/coreApi/createPasteFragment.ts

```diff
@@ -173,3 +176,24 @@ function createBeforePasteEvent(core: EditorCore, clipboardData: ClipboardData):
 function processStyles(node: ParentNode, callback: (style: HTMLStyleElement) => void) {
     toArray(node.querySelectorAll('style')).forEach(callback);
 }
+
+function validateClipboardImageSource(event: BeforePasteEvent, pasteAsText: boolean): boolean {
```

Comment:
We already have code to do check for powerpoint in Paste plugin. In this function it should be neutral and do not do check for any specific scenario.

Reply:
See https://github.com/microsoft/roosterjs/blob/658dd377f31941b7422e1706d1ffe6fc65c2a72d/packages/roosterjs-editor-plugins/lib/plugins/Paste/pptConverter/convertPastedContentFromPowerPoint.ts#L18
It will rebuild the fragment to use HTML when paste from PPT.

Reply:
createPasteFragment() only handles common cases, which means it should not know any special knowledge, just provide whatever it can provide such as html string, image source, ... and provide a document fragment with default paste result. Then let plugin decide what should be pasted finally. Plugin can overwrite the default result.
People can use our Paste plugin, and can also create their own plugin to decide what to paste.